# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Barrier.h
# Commit: 6f7514b0a657
# Full Hash: 6f7514b0a657f97bf3e0e097c2b3faaea142450b
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 2 - Remove owned BaseShapes. r=jonco
#   
#   At this point all BaseShapes are unowned (= immutable, shared) so we can remove the
#   UnownedBaseShape type and the `BaseShape::unowned_` field.
#   
# ==============================================================================

diff -r 76db3b04dedd -r 6f7514b0a657 js/src/gc/Barrier.h
--- a/js/src/gc/Barrier.h	Fri Mar 05 19:10:04 2021 +0000
+++ b/js/src/gc/Barrier.h	Fri Mar 05 19:10:05 2021 +0000
@@ -1115,7 +1115,6 @@
 class ScriptSourceObject;
 class Shape;
 class BaseShape;
-class UnownedBaseShape;
 class WasmInstanceObject;
 class WasmTableObject;
 
@@ -1137,7 +1136,6 @@
 using GCPtrScript = GCPtr<JSScript*>;
 using GCPtrString = GCPtr<JSString*>;
 using GCPtrShape = GCPtr<Shape*>;
-using GCPtrUnownedBaseShape = GCPtr<UnownedBaseShape*>;
 using GCPtrObjectGroup = GCPtr<ObjectGroup*>;
 using GCPtrValue = GCPtr<Value>;
 using GCPtrId = GCPtr<jsid>;